src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,1638 @@
+        alpha (float, optional): Weighting factor in range (0,1) to balance
+                positive vs negative examples. Default = -1 (no weighting).
+        gamma (float, optional): Exponent of the modulating factor (1 - p_t) to
+                balance easy vs hard examples.
+    Returns:
+        Tensor: The computed loss
+    """
+    probs: Tensor = inputs.sigmoid()
+    ce_loss: Tensor = F.binary_cross_entropy_with_logits(inputs, targets, reduction="none")
+    p_t: Tensor = probs * targets + (1 - probs) * (1 - targets)
+    loss: Tensor = ce_loss * ((1 - p_t) ** gamma)
+
+    if alpha >= 0:
+        alpha_t: Tensor = alpha * targets + (1 - alpha) * (1 - targets)
+        loss = alpha_t * loss
+
+    loss = loss.mean(1).sum() / num_masks
+    return loss
+
+
+def pair_wise_dice_loss(inputs: Tensor, targets: Tensor) -> Tensor:
+    """
+    A pair wise version of the dice loss, see `dice_loss` for usage
+
+    Args:
+        inputs (Tensor): A tensor representing a mask
+        targets (Tensor): A tensor with the same shape as inputs. Stores the binary classification labels for each element in inputs (0 for the negative class and 1 for the positive class).
+
+
+    Returns:
+        Tensor: The computed loss between each pairs
+    """
+    inputs: Tensor = inputs.sigmoid()
+    # TODO this .flatten seems to be unecessary because the shape is 2d
+    inputs: Tensor = inputs.flatten(1)
+    # TODO why 1 is not added to the number to avoid numerator = 0 in edge cases?
+    numerator: Tensor = 2 * torch.einsum("nc,mc->nm", inputs, targets)
+    # using broadcasting to get a [NUM_QUERIES, NUM_CLASSES] matrix
+    denominator: Tensor = inputs.sum(-1)[:, None] + targets.sum(-1)[None, :]
+    loss: Tensor = 1 - (numerator + 1) / (denominator + 1)
+    return loss
+
+
+def pair_wise_sigmoid_focal_loss(inputs: Tensor, targets: Tensor, alpha: float = 0.25, gamma: float = 2.0) -> Tensor:
+    """
+    A pair wise version of the focal loss, see `sigmoid_focal_loss` for usage
+
+    Args:
+        inputs (Tensor): A tensor representing a mask
+        targets (Tensor): A tensor with the same shape as inputs. Stores the binary classification labels for each element in inputs (0 for the negative class and 1 for the positive class).
+
+
+    Returns:
+        Tensor: The computed loss between each pairs
+    """
+    if alpha < 0:
+        raise ValueError(f"alpha must be positive")
+
+    hw: int = inputs.shape[1]
+
+    prob: Tensor = inputs.sigmoid()
+    focal_pos: Tensor = ((1 - prob) ** gamma) * F.binary_cross_entropy_with_logits(
+        inputs, torch.ones_like(inputs), reduction="none"
+    )
+    focal_neg: Tensor = (prob ** gamma) * F.binary_cross_entropy_with_logits(
+        inputs, torch.zeros_like(inputs), reduction="none"
+    )
+
+    focal_pos *= alpha
```

Comment:
Since we are doing it on several lines, let's move the `((1 - prob) ** gamma)` here too and maybe the loss will fit in one line.